src/Confluent.Kafka/IConsumer.cs

```diff
@@ -271,6 +271,89 @@ public interface IConsumer<TKey, TValue> : IClient
         void Assign(IEnumerable<TopicPartition> partitions);
 
 
+        /// <summary>
+        ///     Incrementally add <paramref name="partition" />
+        ///     to the current assignment.
+        /// </summary>
+        /// <param name="partition">
+        ///     The additional partition to consume from.
+        ///     Consumption will resume from the last committed
+        ///     offset, or according to the 'auto.offset.reset'
+        ///     configuration parameter if no offsets have been
+        ///     committed yet.
+        /// </param>
+        void IncrementalAssign(TopicPartition partition);
```

Comment:
Why do we need this variant?
I think it will trick people into calling IncrementalAssign() in a loop, rather than a single call, which is not the way the underlying C API is intended to be use (preferably one *assign call per rebalance).

Reply:
yeah, I agree with that. I was following the assign/unssign variants. i don't think there's a good reason to keep this